backend-lib/src/main/rust/lib.rs

```diff
@@ -530,15 +530,15 @@ pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_importAcc
     let res = catch_unwind(&mut env, |env| {
         let network = parse_network(network_id as u32)?;
         let mut db_data = wallet_db(env, network, db_data)?;
-        let ufvk_str = java_string_to_rust(env, &ufvk_str);
+        let ufvk_str = java_string_to_rust(env, &ufvk_str)?;
         let ufvk = UnifiedFullViewingKey::decode(&network, &ufvk_str).map_err(|e| {
             anyhow!(
                 "Value \"{}\" did not decode as a valid UFVK: {}",
                 ufvk_str,
                 e
             )
         })?;
```

Comment:
There are three near-identical uses of this pattern (one uses `map_err` rather than `match`). Add a `ufvk_from_jni` helper.